charts/merlin/templates/deployment.yaml

```diff
@@ -79,17 +79,26 @@ spec:
             value: "{{ .Values.merlin.loggerDestinationURL }}"
           - name: PORT
             value: "{{ .Values.merlin.service.internalPort }}"
+          {{- if .Values.postgresql.enabled }}
           - name: DATABASE_HOST
             value: {{ template "postgresql.host" . }}
           - name: DATABASE_USER
             value: "{{ .Values.postgresql.postgresqlUsername }}"
+          - name: DATABASE_NAME
+            value: "{{ .Values.postgresql.postgresqlDatabase }}"
+          {{- else -}}
+          - name: DATABASE_HOST
+            value: "{{ .Values.merlin.database.host }}"
+          - name: DATABASE_USER
+            value: "{{ .Values.merlin.database.user }}"
+          - name: DATABASE_NAME
+            value: "{{ .Values.merlin.database.name }}"
+          {{- end }}
           - name: DATABASE_PASSWORD
```

Comment:
How are you planning to configure the `DATABASE_PASSWORD` ?
It seems to me this part is still reusing the built-in postgres.

Reply:
I prefer to user to specify the source entirely:
```
            valueFrom:
              secretKeyRef:
                name: {{ .Values.merlin.database.password.secretName }}
                key:  {{ .Values.merlin.database.password.secretKey }}
```
Can you check with any open source helm chart whether this is best practise.

Reply:
I tried to look at various open source, but not found any resemblance except for feast.

For Feast, there is an option to use an existing secret (which deployed internally and separated resources with the helm chart)

I have refactored the database password, rather than using postgres one, have added your code snippets and create a secret resources should postgres is not enabled